### PR TITLE
r/service: Add `external_traffic_policy` to spec

### PR DESCRIPTION
**Pull request by phamann**
This adds the [`externalTrafficPolicy`](https://kubernetes.io/docs/api-reference/v1.7/#servicespec-v1-core) property to the spec schema on the Service resource.

By default k8s obscures the source IP from packets replacing it with the node IP. However certain applications require source IP preservation. This behaviour is managed via the `externalTrafficPolicy`. More info here: https://kubernetes.io/docs/tutorials/services/source-ip/

Until recently we got around the lack of this property by using an internal k8s annotation to apply the behaviour using `service.metadata.annotations[0]`. However, #50 (and the 1.0.0 release) broke this by treating internal annotations as invalid. Using the `spec.externalTrafficPolicy` is the preferred method anyway hence this PR.

### CODE REVIEW

File: kubernetes/structure_service_spec.go [hunk 1]
```diff
@@ -136,6 +136,9 @@ func expandServiceSpec(l []interface{}) v1.ServiceSpec {
 	if v, ok := in["external_name"].(string); ok {
 		obj.ExternalName = v
 	}
+	if v, ok := in["external_traffic_policy"].(string); ok {
+		obj.ExternalName = v
```
Comment (sl1pm4t) [inline kubernetes/structure_service_spec.go]: Is this corrrect - `obj.ExternalName` ?
Shouldn't that be something like `obj.ExternalTrafficPolicy` ?

Comment (radeksimko) [reply kubernetes/structure_service_spec.go]: We'll also need to modify `flattenServiceSpec` similarly.

**Comment by phamann** [comment]
Thanks for the review. I've now updated/added all of the correct references within `flattenServiceSpec()`, `expandServiceSpec()`, and `patchServiceSpec()` in https://github.com/terraform-providers/terraform-provider-kubernetes/pull/59/commits/6bccc7c0f346c417548a7319bbd74affce2cd2f4. 

The build is now failing (correctly) due to the vendored K8S not having the the field. As @radeksimko suggested, I'll wait until the upgrade has been done and then re-run.

**Comment by phamann** [comment]
@radeksimko Any update on when you will upgrade the vendored K8S to 1.7? To resolve the issues with this PR?

**Comment by sl1pm4t** [comment]
@phamann I created a PR #80 that bumps the vendored k8s client code to v1.7.
Do you want to try rebase this PR on mine to see if the build is successful?

**Comment by nicknovitski** [comment]
Since #80 has been merged, it seems this is due to be rebased.

**Comment by phamann** [comment]
@radeksimko Apologies for neglecting this this, we had the provider version pinned as a temporary resolution to this problem internally. Needing some other features in the latest release reminded me of this PR.

I've now rebased with master and the tests are all passing 😄

**Comment by phamann** [comment]
Any update on this @radeksimko? It still has the `waiting-response` label even though I have rebased and fixed issues, which may be why you haven't reviewed again?

**Comment by jhoblitt** [comment]
I just smacked into this as well: `Error: module.pkgroot.kubernetes_service.ssl_proxy: spec.0: invalid or unknown key: externalTrafficPolicy` -- it would be great to see this feature merged.

**Comment by codercr** [comment]
This is good but should default to the Kubernetes default of `Cluster`. We should set to `Local` when one wishes to change default behavior.

**Comment by yacut** [comment]
@phamann @radeksimko any update on this?

**Comment by ulm0** [comment]
Hi, guys. any update on this? i need this field to deploy nginx ingress controller with terraform.

**Comment by annaGo** [comment]
I need this field as well, going to pass on terraform and use kubectl :(

**Comment by wackledude** [comment]
@radeksimko this is a crucial element to deploy nginx ingress controller with terraform as previously mentioned. is there any timeline on this?

**Comment by zioproto** [comment]
I rebased this PR on top of the origin master and I pushed the branch here:
https://github.com/zioproto/terraform-provider-kubernetes/tree/PR-59
@phamann could you check if the rebase is OK for you and push the changes to this PR ?
thanks

**Comment by zioproto** [comment]
@phamann if my rebase looks correct, to solve the git stuff, you could do like this:
```
git remote add zioproto https://github.com/zioproto/terraform-provider-kubernetes
git fetch --all
git checkout b-service-external-traffic-policy
git reset --hard zioproto/PR-59
git push -f origin b-service-external-traffic-policy
```
This will refresh the current PR59 with the rebased version I proposed.
Let me know if you are still interested in getting this merged, otherwise I will propose a new PR.
Thanks !! :)

**Comment by regner** [comment]
Thanks for your work on this everyone. Really looking forward to taking advantage of it.

**Pull request merged**